### PR TITLE
Run RN List Component Jest UTs

### PR DESCRIPTION
**Pull request by NickGerleman**
This enables running the built-in JS component tests for List. We need to make source copies of these since they are not published in the npm package, but we can keep them synchronized via override tooling.

Need to add for react-native-win32 in a later change

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/8070)

### CODE REVIEW

File: vnext/jest.config.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+module.exports = {
+  resolver: './jest-resolver.js',
```
Comment (NickGerleman) [inline vnext/jest.config.js:2]: Should move these to `src/jest` to merge with stock utils and avoid any more root-level pollution.

Comment (NickGerleman) [reply vnext/jest.config.js:2]: I don't agree with my previous opinion.

Having a common pattern among packages like "jest-resolver.js" seems better than special casing to RN stock.

**Review by jonthysell — APPROVED** [review]
Though all the tasks passed it looks like there was a failure in the logs:

https://dev.azure.com/ms/react-native-windows/_build/results?buildId=182692&view=logs&j=b8115773-854b-5a24-9853-4feb4834ad88&t=db6c783b-f947-5190-0fae-f0584c7a302c&l=455

```
2021-06-20T05:34:20.5447830Z verb react-native-windows test |   PASS  Libraries/Lists/__tests__/VirtualizedList-test.js (41.586 s)
2021-06-20T05:34:20.7284510Z verb react-native-windows test |    ● Console
2021-06-20T05:34:20.7385130Z verb react-native-windows test |  
2021-06-20T05:34:20.7488020Z verb react-native-windows test     console.error
2021-06-20T05:34:20.7588220Z verb react-native-windows test |        The above error occurred in the <CellRenderer> component:
2021-06-20T05:34:20.7691050Z verb react-native-windows test |        
2021-06-20T05:34:20.7793570Z verb react-native-windows test |            at CellRenderer (/Users/runner/work/1/s/vnext/Libraries/Lists/VirtualizedList.js:1936:164)
2021-06-20T05:34:20.7894100Z verb react-native-windows test |            at View
2021-06-20T05:34:20.7991220Z verb react-native-windows test |            at View (/Users/runner/work/1/s/vnext/jest/mockComponent.js:28:154)
2021-06-20T05:34:20.8093990Z verb react-native-windows test |            at RCTScrollView
2021-06-20T05:34:20.8196940Z verb react-native-windows test |            at RCTScrollView (/Users/runner/work/1/s/vnext/jest/mockNativeComponent.js:15:156)
2021-06-20T05:34:20.8297630Z verb react-native-windows test |            at ScrollView (/Users/runner/work/1/s/vnext/jest/mockScrollView.js:22:158)
2021-06-20T05:34:20.8400530Z verb react-native-windows test |            at VirtualizedListContextProvider (/Users/runner/work/1/s/vnext/Libraries/Lists/VirtualizedListContext.js:98:23)
2021-06-20T05:34:20.8502370Z verb react-native-windows test |            at VirtualizedList (/Users/runner/work/1/s/vnext/Libraries/Lists/VirtualizedList.js:680:73)
2021-06-20T05:34:20.8603820Z verb react-native-windows test |        
2021-06-20T05:34:20.8706770Z verb react-native-windows test |        Consider adding an error boundary to your tree to customize error handling behavior.
2021-06-20T05:34:20.8806960Z verb react-native-windows test |        Visit https://reactjs.org/link/error-boundaries to learn more about error boundaries.
2021-06-20T05:34:20.8909870Z verb react-native-windows test |  
2021-06-20T05:34:20.9011600Z verb react-native-windows test |        75 |         return;
2021-06-20T05:34:20.9113130Z verb react-native-windows test |        76 |       }
2021-06-20T05:34:20.9214920Z verb react-native-windows test |      > 77 |       consoleError(message);
2021-06-20T05:34:20.9317100Z verb react-native-windows test |           |       ^
2021-06-20T05:34:20.9319440Z verb react-native-windows test |        78 |     });
2021-06-20T05:34:20.9418990Z verb react-native-windows test |        79 |
2021-06-20T05:34:20.9523050Z verb react-native-windows test |        80 |     const componentFactory = () =>
2021-06-20T05:34:20.9623240Z verb react-native-windows test |  
2021-06-20T05:34:20.9668280Z verb react-native-windows test |        at console.<anonymous> (Libraries/Lists/__tests__/VirtualizedList-test.js:77:7)
2021-06-20T05:34:20.9771630Z verb react-native-windows test |        at logCapturedError (../node_modules/react-test-renderer/cjs/react-test-renderer.development.js:10989:27)
2021-06-20T05:34:20.9827560Z verb react-native-windows test |        at update.callback (../node_modules/react-test-renderer/cjs/react-test-renderer.development.js:11022:9)
2021-06-20T05:34:20.9930780Z verb react-native-windows test |        at callCallback (../node_modules/react-test-renderer/cjs/react-test-renderer.development.js:3662:16)
2021-06-20T05:34:21.0031040Z verb react-native-windows test |        at commitUpdateQueue (../node_modules/react-test-renderer/cjs/react-test-renderer.development.js:3683:13)
2021-06-20T05:34:21.0091130Z verb react-native-windows test |  
2021-06-20T05:34:21.0192240Z verb react-native-windows test     console.error
2021-06-20T05:34:21.0294610Z verb react-native-windows test |        The above error occurred in the <CellRenderer> component:
2021-06-20T05:34:21.0395670Z verb react-native-windows test |        
2021-06-20T05:34:21.0435000Z verb react-native-windows test |            at CellRenderer (/Users/runner/work/1/s/vnext/Libraries/Lists/VirtualizedList.js:1936:164)
2021-06-20T05:34:21.0538710Z verb react-native-windows test |            at View
2021-06-20T05:34:21.0639300Z verb react-native-windows test |            at View (/Users/runner/work/1/s/vnext/jest/mockComponent.js:28:154)
2021-06-20T05:34:21.0741980Z verb react-native-windows test |            at RCTScrollView
2021-06-20T05:34:21.0842920Z verb react-native-windows test |            at RCTScrollView (/Users/runner/work/1/s/vnext/jest/mockNativeComponent.js:15:156)
2021-06-20T05:34:21.0909380Z verb react-native-windows test |            at ScrollView (/Users/runner/work/1/s/vnext/jest/mockScrollView.js:22:158)
2021-06-20T05:34:21.1013550Z verb react-native-windows test |            at VirtualizedListContextProvider (/Users/runner/work/1/s/vnext/Libraries/Lists/VirtualizedListContext.js:98:23)
2021-06-20T05:34:21.1031480Z verb react-native-windows test |            at VirtualizedList (/Users/runner/work/1/s/vnext/Libraries/Lists/VirtualizedList.js:680:73)
2021-06-20T05:34:21.1120000Z verb react-native-windows test |        
2021-06-20T05:34:21.1136770Z verb react-native-windows test |        Consider adding an error boundary to your tree to customize error handling behavior.
2021-06-20T05:34:21.1226220Z verb react-native-windows test |        Visit https://reactjs.org/link/error-boundaries to learn more about error boundaries.
2021-06-20T05:34:21.1242200Z verb react-native-windows test |  
2021-06-20T05:34:21.1333350Z verb react-native-windows test |        75 |         return;
2021-06-20T05:34:21.1352120Z verb react-native-windows test |        76 |       }
2021-06-20T05:34:21.1421710Z verb react-native-windows test |      > 77 |       consoleError(message);
2021-06-20T05:34:21.1482880Z verb react-native-windows test |           |       ^
2021-06-20T05:34:21.1539760Z verb react-native-windows test |        78 |     });
2021-06-20T05:34:21.1642570Z verb react-native-windows test |        79 |
2021-06-20T05:34:21.1644230Z verb react-native-windows test |        80 |     const componentFactory = () =>
2021-06-20T05:34:21.1692490Z verb react-native-windows test |  
2021-06-20T05:34:21.1713820Z verb react-native-windows test |        at console.<anonymous> (Libraries/Lists/__tests__/VirtualizedList-test.js:77:7)
2021-06-20T05:34:21.1816980Z verb react-native-windows test |        at logCapturedError (../node_modules/react-test-renderer/cjs/react-test-renderer.development.js:10989:27)
2021-06-20T05:34:21.1920580Z verb react-native-windows test |        at update.callback (../node_modules/react-test-renderer/cjs/react-test-renderer.development.js:11022:9)
2021-06-20T05:34:21.1945640Z verb react-native-windows test |        at callCallback (../node_modules/react-test-renderer/cjs/react-test-renderer.development.js:3662:16)
2021-06-20T05:34:21.1969760Z verb react-native-windows test |        at commitUpdateQueue (../node_modules/react-test-renderer/cjs/react-test-renderer.development.js:3683:13)
2021-06-20T05:34:21.2050380Z verb react-native-windows test |  
2021-06-20T05:34:21.2075790Z verb react-native-windows test     console.error
2021-06-20T05:34:21.2178580Z verb react-native-windows test |        The above error occurred in the <CellRenderer> component:
2021-06-20T05:34:21.2235740Z verb react-native-windows test |        
2021-06-20T05:34:21.2338580Z verb react-native-windows test |            at CellRenderer (/Users/runner/work/1/s/vnext/Libraries/Lists/VirtualizedList.js:1936:164)
2021-06-20T05:34:21.2437050Z verb react-native-windows test |            at View
2021-06-20T05:34:21.2541800Z verb react-native-windows test |            at View (/Users/runner/work/1/s/vnext/jest/mockComponent.js:28:154)
2021-06-20T05:34:21.2573160Z verb react-native-windows test |            at RCTScrollView
2021-06-20T05:34:21.2658620Z verb react-native-windows test |            at RCTScrollView (/Users/runner/work/1/s/vnext/jest/mockNativeComponent.js:15:156)
2021-06-20T05:34:21.2685600Z verb react-native-windows test |            at ScrollView (/Users/runner/work/1/s/vnext/jest/mockScrollView.js:22:158)
2021-06-20T05:34:21.2793510Z verb react-native-windows test |            at VirtualizedListContextProvider (/Users/runner/work/1/s/vnext/Libraries/Lists/VirtualizedListContext.js:98:23)
2021-06-20T05:34:21.2892200Z verb react-native-windows test |            at VirtualizedList (/Users/runner/work/1/s/vnext/Libraries/Lists/VirtualizedList.js:680:73)
2021-06-20T05:34:21.2998040Z verb react-native-windows test |        
2021-06-20T05:34:21.3107030Z verb react-native-windows test |        Consider adding an error boundary to your tree to customize error handling behavior.
2021-06-20T05:34:21.3313630Z verb react-native-windows test |        Visit https://reactjs.org/link/error-boundaries to learn more about error boundaries.
2021-06-20T05:34:21.3464690Z verb react-native-windows test |  
2021-06-20T05:34:21.3567450Z verb react-native-windows test |        75 |         return;
2021-06-20T05:34:21.3667740Z verb react-native-windows test |        76 |       }
2021-06-20T05:34:21.3772950Z verb react-native-windows test |      > 77 |       consoleError(message);
2021-06-20T05:34:21.3873150Z verb react-native-windows test |           |       ^
2021-06-20T05:34:21.3901030Z verb react-native-windows test |        78 |     });
2021-06-20T05:34:21.4003590Z verb react-native-windows test |        79 |
2021-06-20T05:34:21.4044780Z verb react-native-windows test |        80 |     const componentFactory = () =>
2021-06-20T05:34:21.4096540Z verb react-native-windows test |  
2021-06-20T05:34:21.4126520Z verb react-native-windows test |        at console.<anonymous> (Libraries/Lists/__tests__/VirtualizedList-test.js:77:7)
2021-06-20T05:34:21.4239350Z verb react-native-windows test |        at logCapturedError (../node_modules/react-test-renderer/cjs/react-test-renderer.development.js:10989:27)
2021-06-20T05:34:21.4343080Z verb react-native-windows test |        at update.callback (../node_modules/react-test-renderer/cjs/react-test-renderer.development.js:11022:9)
2021-06-20T05:34:21.4446340Z verb react-native-windows test |        at callCallback (../node_modules/react-test-renderer/cjs/react-test-renderer.development.js:3662:16)
2021-06-20T05:34:21.4550070Z verb react-native-windows test |        at commitUpdateQueue (../node_modules/react-test-renderer/cjs/react-test-renderer.development.js:3683:13)
2021-06-20T05:34:21.4652750Z verb react-native-windows test |  
2021-06-20T05:34:21.4755270Z verb react-native-windows test     console.warn
2021-06-20T05:34:21.4857630Z verb react-native-windows test |        VirtualizedList: missing keys for items, make sure to specify a key or id property on each item or provide a custom keyExtractor. 
2021-06-20T05:34:21.4958070Z verb react-native-windows test |  
2021-06-20T05:34:21.5059050Z verb react-native-windows test |        1875 |      * delete this comment and run Flow. */
2021-06-20T05:34:21.5161160Z verb react-native-windows test |        1876 |     return frame;
2021-06-20T05:34:21.5262270Z verb react-native-windows test |      > 1877 |   };
2021-06-20T05:34:21.5365160Z verb react-native-windows test |             |               ^
2021-06-20T05:34:21.5465410Z verb react-native-windows test |        1878 |
2021-06-20T05:34:21.5565600Z verb react-native-windows test |        1879 |   _updateViewableItems(data: any) {
2021-06-20T05:34:21.5668400Z verb react-native-windows test |        1880 |     const {getItemCount} = this.props;
2021-06-20T05:34:21.5763520Z verb react-native-windows test |  
2021-06-20T05:34:21.5794310Z verb react-native-windows test |        at VirtualizedList.render (Libraries/Lists/VirtualizedList.js:1877:14266)
2021-06-20T05:34:21.5897470Z verb react-native-windows test |        at finishClassComponent (../node_modules/react-test-renderer/cjs/react-test-renderer.development.js:8459:35)
2021-06-20T05:34:21.5904300Z verb react-native-windows test |        at updateClassComponent (../node_modules/react-test-renderer/cjs/react-test-renderer.development.js:8409:28)
2021-06-20T05:34:21.6007090Z verb react-native-windows test |        at beginWork (../node_modules/react-test-renderer/cjs/react-test-renderer.development.js:9990:20)
2021-06-20T05:34:21.6070330Z verb react-native-windows test |  (../node_modules/react-test-renderer/cjs/react-test-renderer.development.js:13800:16)
2021-06-20T05:34:21.6173150Z verb react-native-windows test |        at workLoopSync (../node_modules/react-test-renderer/cjs/react-test-renderer.development.js:13728:9)
2021-06-20T05:34:21.6275850Z verb react-native-windows test |        at renderRootSync (../node_modules/react-test-renderer/cjs/react-test-renderer.development.js:13691:11)
2021-06-20T05:34:21.6367900Z verb react-native-windows test |        at performSyncWorkOnRoot (../node_modules/react-test-renderer/cjs/react-test-renderer.development.js:13404:22)
2021-06-20T05:34:21.6454220Z verb react-native-windows test |   PASS  Libraries/Lists/__tests__/FlatList-test.js
2021-06-20T05:34:21.6556990Z verb react-native-windows test |   PASS  Libraries/Lists/__tests__/FillRateHelper-test.js
2021-06-20T05:34:21.7484820Z verb react-native-windows test |   PASS  Libraries/Lists/__tests__/VirtualizeUtils-test.js
2021-06-20T05:34:21.9494000Z verb react-native-windows test |   PASS  Libraries/Lists/__tests__/SectionList-test.js
2021-06-20T05:34:22.0158190Z verb react-native-windows test |  Test Suites: 7 passed, 7 total
2021-06-20T05:34:22.0160130Z verb react-native-windows test |  Tests:       77 passed, 77 total
2021-06-20T05:34:22.0161950Z verb react-native-windows test |  Snapshots:   41 passed, 41 total
2021-06-20T05:34:22.0163670Z verb react-native-windows test |  Time:        43.488 s
2021-06-20T05:34:22.0165230Z verb react-native-windows test |  Ran all test suites.
2021-06-20T05:34:22.0933380Z verb react-native-windows test |  hash put 938f60a444d6c64efd264125fec027f221a1c118
2021-06-20T05:34:22.9502660Z info react-native-windows test ✔️ done  - 46.21s
```

**Comment by NickGerleman** [comment]
@jonthysell I believe the error is expected for the test, but the component still writes to console.

The builtin RN tests are a little bit spammy with console output I've noticed, but I think it's ignorable.

File: vnext/jest.config.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+module.exports = {
```
Comment (NickGerleman) [inline vnext/jest.config.js:1]: Make verbose for consistency with other tests

**Pull request merged**